src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager_Blocking.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Protocols.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.IdentityModel.Telemetry;
+
+namespace Microsoft.IdentityModel.Protocols
+{
+    partial class ConfigurationManager<T> where T : class
+    {
+        private readonly SemaphoreSlim _refreshLock = new(1);
+
+        private TimeSpan _bootstrapRefreshInterval = TimeSpan.FromSeconds(1);
+
+        private async Task<T> GetConfigurationWithBlockingAsync(CancellationToken cancel)
+        {
+            Exception _fetchMetadataFailure = null;
+            await _refreshLock.WaitAsync(cancel).ConfigureAwait(false);
```

Comment:
8.0.1 has the opportunistic check on time to refresh before the lock. Is the re-ordering intentional (e.g. sync after update)?

Reply:
It was moved to the call site.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/4458df8390452d53bbaac6406908bf6c6dd96e69/src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs#L209